internal/app/machined/main.go

```diff
@@ -157,31 +158,48 @@ func main() {
 			rebootFlag = unix.LINUX_REBOOT_CMD_POWER_OFF
 			fallthrough
 		case event.Reboot:
-			if err := seq.Shutdown(); err != nil {
-				panic(fmt.Errorf("shutdown failed: %w", err))
-			}
+			go func() {
+				if err := seq.Shutdown(); err != nil {
```

Comment:
this flow still allows for (slightly) concurrently sequences, as when `seq.Shutdown()` returns, another event might launch another sequence while this goroutine progresses through the `sync()` and other steps

while what I proposed below might be a more robust solution (if we push startup as event I guess)